ngx_http_geoip2_module.c

```diff
@@ -394,6 +417,44 @@ ngx_http_geoip2_add_variable(ngx_conf_t *cf, ngx_command_t *dummy, void *conf)
     name.data++;
     nelts = (int) cf->args->nelts;
     idx = 1;
+
+    if (nelts > 0 && value[1].len == 8 && ngx_strncmp(value[1].data, "metadata", 8) == 0) {
+        if (nelts != 3) {
```

Comment:
this function is getting pretty long and the purpose of this `if` block is separate from the following code; I'd probably move it into a new function unless the code above makes it too painful